src/mprk.jl

```diff
@@ -185,12 +144,13 @@ as keyword argument `linsolve`.
   Applied Numerical Mathematics 47.1 (2003): 1-30.
   [DOI: 10.1016/S0168-9274(03)00101-6](https://doi.org/10.1016/S0168-9274(03)00101-6)
 """
-struct MPE{F} <: OrdinaryDiffEqAlgorithm
+struct MPE{F, T} <: OrdinaryDiffEqAlgorithm
     linsolve::F
+    small_constant::T
 end
 
-function MPE(; linsolve = LUFactorization())
-    MPE(linsolve)
+function MPE(; linsolve = LUFactorization(), small_constant = floatmin())
+    MPE(linsolve, small_constant)
 end
```

Comment:
This is a bit problematic. Before, we have used `floatmin(uEltypeNoUnits)` when constructing the cache. Thus, the floating point type was adapted to the type we use for the computations. Now, we can get something like

```julia
julia> u = zeros(Float32, 2)
2-element Vector{Float32}:
 0.0
 0.0

julia> @. u += floatmin()
2-element Vector{Float32}:
 0.0
 0.0
```

Thus, I propose to use a function in the algorithm that returns the concrete value given the type of numbers we're using, i.e.,

```suggestion
function MPE(; linsolve = LUFactorization(), small_constant = floatmin)
    MPE(linsolve, small_constant)
end
```

Reply:
I see that my suggestion is not working, but I find something like `small_constant = Returns(1.0e-50)`  not very user-friendly, as the variable name suggests it should be a number.

What if we use `small_constant=nothing` by default for the algorithm and then initialize it  with `floatmin(uEltypeNoUnits)` in the constants cache? And if the algorithm is called with a specific value for `small_constant` e.g. `SSPMPRK43(small_constant=1e-50)` we use this specific value?

Reply:
Yes, that's a good suggestion, too 👍 We could have something like the following setup:

```suggestion
function MPE(; linsolve = LUFactorization(), small_constant = nothing)
    if isnothing(small_constant)
        small_constant_function = floatmin
    elseif small_constant isa Number
        small_constant_function = Returns(small_constant)
    else # assume small_constant isa Function
        small_constant_function = small_constant
    end
    MPE(linsolve, small_constant_function)
end
```